accounts.go

```diff
@@ -0,0 +1,94 @@
+/* Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+package keyhub
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/dghubble/sling"
+	"github.com/topicuskeyhub/go-keyhub/model"
+)
+
+type AccountService struct {
+	sling *sling.Sling
+}
+
+func newAccountService(sling *sling.Sling) *AccountService {
+	return &AccountService{
+		sling: sling.Path("/keyhub/rest/v1/account/"),
+	}
+}
+
+func (s *AccountService) List() (accounts []model.Account, err error) {
+	results := new(model.AccountList)
+	errorReport := new(model.ErrorReport)
+
+	_, err = s.sling.New().Get("").Receive(results, errorReport)
+	if errorReport.Code > 0 {
+		err = errors.New("Could not get Accounts. Error: " + errorReport.Message)
+	}
+	if err == nil {
+		if len(results.Items) > 0 {
+			accounts = results.Items
+		} else {
+			accounts = []model.Account{}
+		}
+	}
+
+	return
+}
+
+func (s *AccountService) GetByUUID(uuid string) (result *model.Account, err error) {
```

Comment:
uuid validatie voordat het op de lijn gaat? mss zelfs een uuid type gebruiken ipv string (bv https://pkg.go.dev/github.com/google/uuid)?

wat is nou het plan met de keyhub sdk? voor de terraform provider is deze client bijwerken de kortste klap, maar als er een generated sdk voor alle talen gaat komen, hoeveel effort moeten we dan in dit ding steken? Als in, string is fine, deze client gaat toch vervangen worden, of mss moeten we toch wat extra aandacht aan de functie signatures besteden?

Reply:
Ik denk dat we uit deze sdk veel lessen kunnen leren voor de sdk die we willen gaan genereren. Ook denk ik dat het nog wel even kan duren voordat we een platform voor generatie hebben, en deze sdk nog wel een poosje mee moet. Waarschijnlijk is het daarom handig om ook voor deze sdk dit soort dingen netjes te doen.

Reply:
Een uuid type gebruiken is geen gek idee.

Deze SDK moet inderdaad nog wel even meegaan. Ik vind zelf de implementatie van de functions belangrijker, hier kunnen we dingen van leren en verbeteren. Tot zover heeft versie 0.2.x ook lang gewerkt neem ik aan.

Reply:
Een generated SDK gaat niet magisch zijn, er zal altijd handwerk verricht moeten worden. Voor de meeste CRUD zaken moet het wel automatisch lukken maar tot iemand een generator heeft gemaakt is het 100% handwerk.